core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -625,9 +626,10 @@ private void maybeUpdateCopiedOffset(UnifiedLog log) throws RemoteStorageExcepti
                 // of a segment with that epoch copied into remote storage. If it can not find an entry then it checks for the
                 // previous leader epoch till it finds an entry, If there are no entries till the earliest leader epoch in leader
                 // epoch cache then it starts copying the segments from the earliest epoch entry's offset.
-                copiedOffsetOption = OptionalLong.of(findHighestRemoteOffset(topicIdPartition, log));
+                copiedOffsetOption = Optional.of(findHighestRemoteOffset(topicIdPartition, log));
                 logger.info("Found the highest copiedRemoteOffset: {} for partition: {} after becoming leader, " +
                                 "leaderEpoch: {}", copiedOffsetOption, topicIdPartition, leaderEpoch);
+                copiedOffsetOption.ifPresent(epochAndOffset ->  log.updateHighestOffsetInRemoteStorage(epochAndOffset.offset()));
```

Comment:
After broker restart, if there are no more segments to upload, then the `copiedOffset` might be stale. It's good to update the `highestOffsetInRemoteStorage` in the UnifiedLog once we compute it.